Library/Homebrew/download_strategy.rb

```diff
@@ -287,7 +287,7 @@ def fetch
 
       ohai "Downloading #{url}"
 
-      resolved_url, _, url_time = resolve_url_basename_time(url)
+      resolved_url, _, url_time = resolved_url_basename_and_time
```

Comment:
This will break mirrors, since then for every mirror the original URL would be used.

Reply:
I was thinking the same as @reitermarkus: as Hash cache in `resolve_url_basename_time` seems like a good idea.

Reply:
Ok. I'm not sure I entirely follow what you're suggesting, but I committed an attempt to minimize the API changes. I'm also not sure what the advantages of using a Hash to cache multiple resolved urls over just overriding the resolved url on `CurlDownloadStrategyError`. Is there some reason to keep all the information from multiple resolved urls for a single resource?